src/app.js

```diff
@@ -29,66 +29,82 @@ const { initializeRedisClient } = require('./utils/redis');
 const { setupAppHooks } = require('./app.hooks');
 const versionService = require('./services/version');
 
-const app = express(feathers());
-app.disable('x-powered-by');
+const setupApp = async () => {
+	const app = express(feathers());
+	app.disable('x-powered-by');
 
-const config = configuration();
-app.configure(config);
+	const config = configuration();
+	app.configure(config);
 
-if (LEAD_TIME) {
-	app.use((req, res, next) => {
-		req.leadTime = Date.now();
-		next();
-	});
-}
+	if (LEAD_TIME) {
+		app.use((req, res, next) => {
+			req.leadTime = Date.now();
+			next();
+		});
+	}
 
-const metricsOptions = {};
-if (METRICS_PATH) {
-	metricsOptions.metricsPath = METRICS_PATH;
-}
-app.use(apiMetrics(metricsOptions));
+	const metricsOptions = {};
+	if (METRICS_PATH) {
+		metricsOptions.metricsPath = METRICS_PATH;
+	}
+	app.use(apiMetrics(metricsOptions));
 
-setupSwagger(app);
-initializeRedisClient();
-rabbitMq.setup(app);
+	setupSwagger(app);
+	initializeRedisClient();
+	rabbitMq.setup(app);
 
-app
-	.use(compress())
-	.options('*', cors())
-	.use(cors())
-	.use(favicon(path.join(app.get('public'), 'favicon.ico')))
-	.use('/', express.static('public'))
-	.configure(sentry)
-	.use('/helpdesk', bodyParser.json({ limit: BODYPARSER_JSON_LIMIT }))
-	.use('/', bodyParser.json({ limit: '10mb' }))
-	.use(bodyParser.urlencoded({ extended: true }))
-	.use(bodyParser.raw({ type: () => true, limit: '10mb' }))
-	.use(versionService)
-	.use(defaultHeaders)
-	.get('/system_info/haproxy', (req, res) => {
-		res.send({ timestamp: new Date().getTime() });
-	})
-	.get('/ping', (req, res) => {
-		res.send({ message: 'pong', timestamp: new Date().getTime() });
-	})
-	.configure(rest(handleResponseType))
-	.configure(socketio())
-	.configure(requestLogger)
-	.use((req, res, next) => {
-		// pass header into hooks.params
-		// todo: To create a fake requestId on this place is a temporary solution
-		// it MUST be removed after the API gateway is established
-		const uid = ObjectId();
-		req.headers.requestId = uid.toString();
-		req.feathers.leadTime = req.leadTime;
-		req.feathers.headers = req.headers;
-		req.feathers.originalUrl = req.originalUrl;
-		next();
-	})
-	.configure(services)
-	.configure(sockets)
-	.configure(middleware)
-	.configure(setupAppHooks)
-	.configure(errorHandler);
+	// set custom response header for ha proxy
+	if (KEEP_ALIVE) {
```

Comment:
this should be removed as it was removed in a Hotfix

Reply:
hm github didnt show me this message until I clicked merged... I fixed it in a seperate change.